plugins/janus_sipre.c

```diff
@@ -0,0 +1,2934 @@
+		if(item && item->value) {
+			/* Verify that the address is valid */
+			struct ifaddrs *ifas = NULL;
+			janus_network_address iface;
+			janus_network_address_string_buffer ibuf;
+			if(getifaddrs(&ifas) || ifas == NULL) {
+				JANUS_LOG(LOG_ERR, "Unable to acquire list of network devices/interfaces; some configurations may not work as expected...\n");
+			} else {
+				if(janus_network_lookup_interface(ifas, item->value, &iface) != 0) {
+					JANUS_LOG(LOG_WARN, "Error setting local IP address to %s, falling back to detecting IP address...\n", item->value);
+				} else {
+					if(janus_network_address_to_string_buffer(&iface, &ibuf) != 0 || janus_network_address_string_buffer_is_null(&ibuf)) {
+						JANUS_LOG(LOG_WARN, "Error getting local IP address from %s, falling back to detecting IP address...\n", item->value);
+					} else {
+						local_ip = g_strdup(janus_network_address_string_from_buffer(&ibuf));
+					}
+				}
+			}
+		}
+
+		item = janus_config_get_item_drilldown(config, "general", "keepalive_interval");
+		if(item && item->value) {
+			keepalive_interval = atoi(item->value);
+		}
+		JANUS_LOG(LOG_VERB, "SIPre keep-alive interval set to %d seconds\n", keepalive_interval);
+
+		item = janus_config_get_item_drilldown(config, "general", "register_ttl");
+		if(item && item->value) {
+			register_ttl = atoi(item->value);
+		}
+		JANUS_LOG(LOG_VERB, "SIPre registration TTL set to %d seconds\n", register_ttl);
+
+		item = janus_config_get_item_drilldown(config, "general", "behind_nat");
+		if(item && item->value) {
+			behind_nat = janus_is_true(item->value);
+		}
+
+		item = janus_config_get_item_drilldown(config, "general", "user_agent");
+		if(item && item->value) {
+			user_agent = g_strdup(item->value);
+		} else {
+			user_agent = g_strdup("Janus WebRTC Gateway SIPre Plugin "JANUS_SIPRE_VERSION_STRING);
+		}
+		JANUS_LOG(LOG_VERB, "SIPre User-Agent set to %s\n", user_agent);
+
+		item = janus_config_get_item_drilldown(config, "general", "events");
+		if(item != NULL && item->value != NULL) {
+			notify_events = janus_is_true(item->value);
+		}
+		if(!notify_events && callback->events_is_enabled()) {
+			JANUS_LOG(LOG_WARN, "Notification of events to handlers disabled for %s\n", JANUS_SIPRE_NAME);
+		}
+
+		janus_config_destroy(config);
+	}
+	config = NULL;
+
+	if(local_ip == NULL) {
+		local_ip = janus_network_detect_local_ip_as_string(janus_network_query_options_any_ip);
+		if(local_ip == NULL) {
+			JANUS_LOG(LOG_WARN, "Couldn't find any address! using 127.0.0.1 as the local IP... (which is NOT going to work out of your machine)\n");
+			local_ip = g_strdup("127.0.0.1");
+		}
+	}
+	JANUS_LOG(LOG_VERB, "Local IP set to %s\n", local_ip);
+
+#ifdef HAVE_SRTP_2
+	/* Init randomizer (for randum numbers in SRTP) */
+	RAND_poll();
+#endif
+
+	/* Setup libre */
+	int err = libre_init();
+	if(err) {
+		JANUS_LOG(LOG_ERR, "libre_init() failed: %d (%s)\n", err, strerror(err));
+		return -1;
+	}
+	poll_method_set(poll_method_best());
```

Comment:
this should not be needed. if you have to set the polling-method explicitly, it indicates that
something is wrong.

it is done internally in libre main.c:

https://github.com/creytiv/re/blob/master/src/main/main.c#L539

also you need to call `sip_transp_add` from the re-main thread (using mqueue)
otherwise the system will poll the wrong file-descriptor.